File: CS/MainForm.cs
using System.Windows.Forms;
using DevExpress.Utils;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Ribbon;
using System.Reflection;
using System.Drawing;
using DevExpress.XtraBars.ViewInfo;
using DevExpress.XtraBars.Controls;
using DevExpress.XtraBars.Forms;

namespace RibbonApplicationMenuBottomPaneFlicker
{
    public partial class MainForm : RibbonForm
    {
        public MainForm()
        {
            InitializeComponent();
        }

        private void exitButton_ItemClick(object sender, ItemClickEventArgs e)
        {
            Close();
        }

        private void simpleButton1_Click(object sender, System.EventArgs e)
        {
            Close();
        }

        private void simpleButton1_Click_1(object sender, System.EventArgs e)
        {
        }

        bool internalCalculating = false;

        private void applicationMenu_BeforePopup(object sender, System.ComponentModel.CancelEventArgs e)
        {
            if (internalCalculating)
                return;

            internalCalculating = true;

            try
            {
               MethodInfo mi = typeof(ApplicationMenu).GetMethod("CreateSubControl", BindingFlags.NonPublic | BindingFlags.Instance);
               IFormContainedControl pc = mi.Invoke(applicationMenu, new object[] { Ribbon.Manager }) as IFormContainedControl;
               bottomPaneControlContainer.Width = pc.CalcSize(-1, -1).Width;
            }
            finally {
                internalCalculating = false;
            }


        }
    }
}

File: VB/MainForm.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.Utils
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Ribbon
Imports System.Reflection
Imports System.Drawing
Imports DevExpress.XtraBars.ViewInfo
Imports DevExpress.XtraBars.Controls
Imports DevExpress.XtraBars.Forms

Namespace RibbonApplicationMenuBottomPaneFlicker
	Partial Public Class MainForm
		Inherits RibbonForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub exitButton_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) Handles exitButton.ItemClick
			Close()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles exitSimpleButton.Click
			Close()
		End Sub

		Private Sub simpleButton1_Click_1(ByVal sender As Object, ByVal e As System.EventArgs)
		End Sub

		Private internalCalculating As Boolean = False

		Private Sub applicationMenu_BeforePopup(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs) Handles applicationMenu.BeforePopup
			If internalCalculating Then
				Return
			End If

			internalCalculating = True

			Try
			   Dim mi As MethodInfo = GetType(ApplicationMenu).GetMethod("CreateSubControl", BindingFlags.NonPublic Or BindingFlags.Instance)
			   Dim pc As IFormContainedControl = TryCast(mi.Invoke(applicationMenu, New Object() { Ribbon.Manager }), IFormContainedControl)
			   bottomPaneControlContainer.Width = pc.CalcSize(-1, -1).Width
			Finally
				internalCalculating = False
			End Try


		End Sub
	End Class
End Namespace